content/articles/How-to-build-a-location-based-app-using-Django-and-GeoDjango/index.md

```diff
@@ -0,0 +1,240 @@
+```
+import json
+from django.contrib.gis.geos import fromstr
+from pathlib import Path
+```
+Next we will create the load_data() function
+```
+DATA_FILENAME = 'export.json'
+def load_data(apps, schema_editor):
+    Facility = apps.get_model('facilities', 'Facility')
+    jsonfile = Path(__file__).parents[2] / DATA_FILENAME
+
+    with open(str(jsonfile), encoding="utf8") as datafile:
+        objects = json.load(datafile)
+        for obj in objects['elements']:
+            try:
+                objType = obj['type']
+                if objType == 'node':
+                    tags = obj['tags']
+                    name = tags.get('name','no-name')
+                    longitude = obj.get('lon', 0)
+                    latitude = obj.get('lat', 0)
+                    location = fromstr(f'POINT({longitude} {latitude})', srid=4326)
+                    Facility(name=name, location=location).save()
+            except KeyError:
+                pass 
+```
+
+We loop through the objects of the element containing the location and tags shop, inside the loop we extracted the name and coordinates. Then we return a valid GEOSGeometry object that is corresponding to spatial data. The with statement automatically close the file. We will call the function inside the Migration class.
+
+```
+operations = [
+        migrations.RunPython(load_data)
+    ]
+```
+Here is a copy of the migration file
+
+
+    from django.db import migrations  
+    import json  
+    from django.contrib.gis.geos import fromstr  
+    from pathlib import Path  
+    DATA_FILENAME = 'hotels.json'  
+    def load_data(apps, schema_editor):  
+    Hotel = apps.get_model('hotels', 'Hotel')  
+    jsonfile = Path(__file__).parents[2] / DATA_FILENAME  
+ 
+    with open(str(jsonfile)) as datafile:  
+        objects = json.load(datafile)  
+        for obj in objects['elements']:  
+            try:  
+                objType = obj['type']  
+                if objType == 'node':  
+                    tags = obj['tags']  
+                    name = tags.get('name','no-name')  
+                    longitude = obj.get('lon', 0)  
+                    latitude = obj.get('lat', 0)  
+                    location = fromstr(f'POINT({longitude} {latitude})', srid=4326)  
+                    Hotel(name=name, location=location).save()  
+            except KeyError:  
+                pass  
+    class Migration(migrations.Migration):  
+    dependencies = [  
+        ('hotels', '0001_initial'),  
+    ]  
+  
+    operations = [  
+        migrations.RunPython(load_data)  
+    ]
+
+So let's finish the migration by running:
+ ```
+python manage.py migrate
+```
+from the terminal.
+
+You can log in to the admin dashboard to view all the imported hotels.
+
+Right now we will display all the hotels using Django views, we will use the generic ListView class function. Let's open the hotels/views.py file and type some code.
```

Comment:
Here too, kindly check all instances as well

Reply:
I have effected the corrections.